### PR TITLE
Upgrade material-ui to v3, babel to v7, styleguidist to v9

### LINKED ISSUE #2: Doesn't work with latest version of material-ui (v3.x)
I'm getting an error `Uncaught Error: Cannot find module "material-ui/Typography"`.

Per https://material-ui.com/api/typography/ documentation for latest version imports are a bit different, instead of:
```
import { Typography } from 'material-ui'
```
should be:
```
import Typography from '@material-ui/core/Typography';
```

I'm not sure you can make it backwards compatible.

### PR DESCRIPTION
**Pull request by RadomirPerisic**
This upgrades `material-ui` to latest version.

I couldn't get `styleguidist` to run with current setup so I upgraded that as well (also `webpack` and `babel`)

Fixes #2

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Comment (leMaik) [inline babel.config.js]: Since there isn't any logic in here, this file could still be `.babelrc`, but I think that's fine. :man_shrugging:

**Review by leMaik — APPROVED** [review]

**Review by leMaik — APPROVED** [review]
Now this is a great first contribution! :tada:

File: babel.config.js (hunk 1 above)
Comment (saschb2b) [reply babel.config.js]: That's also the only thing I would change

Comment (RadomirPerisic) [reply babel.config.js]: New docs for babel suggest `babel.config.js` instead, but yes, they serve same purpose

Comment (RadomirPerisic) [reply babel.config.js]: 👍 I'll restore old file

Comment (RadomirPerisic) [reply babel.config.js]: Done

**Comment by RadomirPerisic** [comment]
If you think it's ready you can merge. I don't have write access to merge it.

**Review by saschb2b — APPROVED** [review]

File: babel.config.js (hunk 1 above)
Comment (leMaik) [reply babel.config.js]: image

I don't know Guy Fieri, so I'll stick with use case 2. :sweat_smile:

**Pull request merged**